src/snapred/backend/data/LocalDataService.py

```diff
@@ -535,292 +340,155 @@ def writeNormalizationRecord(
         -- assumes that `writeCalibrationRecord` has already been called, and that the version folder exists
         """
-        calibrationDataPath = self._constructCalibrationDataPath(record.runNumber, record.useLiteMode, record.version)
-
-        # Assumes all workspaces are of WNG-type:
-        workspaces = record.workspaces.copy()
-        for wsName in workspaces.pop(wngt.DIFFCAL_OUTPUT, []):
-            # Rebuild the filename to strip any "iteration" number:
-            #   * WARNING: this workaround does not work correctly if there are multiple workspaces of each "unit" type.
-            ext = Config["calibration.diffraction.output.extension"]
-            if wng.Units.DSP.lower() in wsName:
-                filename = Path(
-                    wng.diffCalOutput()
-                    .unit(wng.Units.DSP)
-                    .runNumber(record.runNumber)
-                    .version(record.version)
-                    .group(record.focusGroupCalibrationMetrics.focusGroupName)
-                    .build()
-                    + ext
-                )
-            else:
-                raise RuntimeError(
-                    f"cannot save a workspace-type: {wngt.DIFFCAL_OUTPUT} without a units token in its name {wsName}"
-                )
-            self.writeRaggedWorkspace(calibrationDataPath, filename, wsName)
-        for wsName in workspaces.pop(wngt.DIFFCAL_DIAG, []):
-            print(f"WORKSPACE {wsName}")
-            ext = Config["calibration.diffraction.diagnostic.extension"]
-            if wng.Units.DIAG.lower() in wsName:
-                filename = Path(
-                    wng.diffCalOutput()
-                    .unit(wng.Units.DIAG)
-                    .runNumber(record.runNumber)
-                    .version(record.version)
-                    .group(record.focusGroupCalibrationMetrics.focusGroupName)
-                    .build()
-                    + ext
-                )
-            else:
-                raise RuntimeError(f"Cannot save workspace-type {wngt.DIFFCAL_DIAG} without diagnostic in its name")
-            self.writeWorkspace(calibrationDataPath, filename, wsName)
-            assert (calibrationDataPath / filename).exists()
-        for tableWSName, maskWSName in zip(
-            workspaces.pop(wngt.DIFFCAL_TABLE, []),
-            workspaces.pop(wngt.DIFFCAL_MASK, []),
-        ):
-            # Rebuild the filename to strip any "iteration" number:
-            #   * WARNING: this workaround does not work correctly if there are multiple table workspaces.
-            diffCalFilename = Path(
-                wng.diffCalTable().runNumber(record.runNumber).version(record.version).build() + ".h5"
-            )
-            self.writeDiffCalWorkspaces(
-                calibrationDataPath,
-                diffCalFilename,
-                tableWorkspaceName=tableWSName,
-                maskWorkspaceName=maskWSName,
-            )
```

Comment:
As above, this rename logic was moved to the `CalibrationService`

Reply:
There still needs to be a `writeWorkspaces` method.  The `CalibrationService` shouldn't need to figure out how to save the workspaces -- that's still `LocalDataService`'s responsibility.  However, it _does_ need to set the workspace names so that they are correct (both in the record, and possibly also on the ADS-resident workspaces themselves).